rogue-thi-app/pages/food.js

```diff
@@ -104,222 +125,323 @@ export default function Mensa () {
+              </p>
+            </Modal.Body>
+
+            <Modal.Footer>
+              <Button variant="primary" onClick={() => setShowMealDetails(null)}>OK</Button>
+            </Modal.Footer>
+          </Modal>
+
+          <Modal show={showFilterSelection} onHide={() => setShowFilterSelection(false)}>
+            <Modal.Header closeButton>
+              <Modal.Title>Filter</Modal.Title>
+            </Modal.Header>
+
+            <Modal.Body>
+              <Modal.Header>
+                Restaurant
+              </Modal.Header>
+
+              <Form>
+                <Modal.Body>
+                  <Form.Check
+                      id={'restaurant-checkbox-mensa'}
+                      label={<span><strong>Mensa</strong></span>}
+                      checked={selectedRestaurants.includes('mensa')}
+                      onChange={() => toggleSelectedRestaurant('mensa')}
+                  />
+                  <Form.Check
+                      id={'restaurant-checkbox-reimanns'}
+                      label={<span><strong>Reimanns</strong></span>}
+                      checked={selectedRestaurants.includes('reimanns')}
+                      onChange={() => toggleSelectedRestaurant('reimanns')}
+                  />
+                </Modal.Body>
+              </Form>
+
+              <Modal.Header>
+                  <div>
+                    Allergene
+                  </div>
+                  <div>
+                    <Button variant={'outline-primary'} onClick={() => setShowAllergenSelection(true)}>
+                      <FontAwesomeIcon title="Allergene" icon={faAllergies} fixedWidth />
+                    </Button>
+                  </div>
+              </Modal.Header>
+
+              <Modal.Body>
+                <span>
+                  {Object.entries(allergenMap).map(([mapkey, mapvalue]) => (
+                    Object.entries(allergenSelection).map(([key, value]) => (
+                          <span key={key} >{mapkey === key && value === true && mapkey + ' - ' + mapvalue + ', '}</span>
+                    ))
+                  ))
+                  }
+                </span>
+              </Modal.Body>
+
+              <Modal.Header>
+                <div>
+                  Essenspräferenzen
+                </div>
+                <div>
+                  <Button variant={'outline-primary'} onClick={() => setShowPreferencesSelection(true)}>
+                    <FontAwesomeIcon title='Preferences' icon={faBowlRice} fixedWidth />
+                  </Button>
+                </div>
+              </Modal.Header>
+
+              <Modal.Body>
+                <span>
+                  {Object.entries(flagMap).map(([mapkey, mapvalue]) => (
+                    Object.entries(preferencesSelection).map(([key, value]) => (
+                          <span key={key} >{mapkey === key && value === true && mapvalue + ', '}</span>
+                    ))
+                  ))
+                  }
```

Comment:
See above comment regarding allergeneMap and allergeneSelection

Reply:
Will be fixed in next push.